File: striver_A2Z_sheet/arrays/easy/S2.java
public class S2 {
    // GFG
    int print2largest(int arr[], int n) {
        int largest = arr[0];
        int secLargest = Integer.MIN_VALUE;
        for(int i: arr)
        {
            if (i > largest)
            {
                secLargest = largest;
                largest = i;
            }
            if (i > secLargest && i != largest)
            {
                secLargest = i;
            }
        }
        if (secLargest == Integer.MIN_VALUE)
        {
            return -1;
        }
        else
        {
            return secLargest;
        }
    }
    // Striver; Given an array, find the second smallest and second largest element in the array.
    // Print ‘-1’ in the event that either of them doesn’t exist.
    // smallest has same logic just "<" instead of ">"
}

File: striver_A2Z_sheet/arrays/rust/easy/S2.rs
// function to get second largest element of array

fn getSecLar(arr: &[i32]) -> i32 { // works with Slices (&arr) and Vec<i32> as &vec.
    let mut lar : i32 = i32::MIN;
    let mut secLar : i32 = i32::MIN;
    for x in arr.iter() {
        if *x > lar {
            secLar = lar;
            lar = *x;
        }
        if *x > secLar && *x != lar {
            secLar = *x;
        }
    }
    return secLar;
}